packages/playwright-core/src/server/browserType.ts

```diff
@@ -272,7 +272,9 @@ export abstract class BrowserType extends SdkObject {
     progress.cleanupWhenAborted(() => closeOrKill(progress.timeUntilDeadline()));
     const wsEndpoint = (await readyState?.waitUntilReady())?.wsEndpoint;
     if (options.cdpPort !== undefined || !this.supportsPipeTransport()) {
-      transport = await WebSocketTransport.connect(progress, wsEndpoint!);
+      if (!wsEndpoint)
```

Comment:
I don't think this is related to wsEndpoint. We should rather throw if the browser is launched with the same profile more than once. Saying "Browser with the given profile is already running".